lib/neo4j/active_node/persistence.rb

```diff
@@ -158,12 +155,19 @@ def load_entity(id)
 
       private
 
+      def find_or_create_query(find_attributes, set_attributes = {})
+        on_create_attributes = set_attributes.reverse_merge(on_create_props(find_attributes))
+
+        neo4j_session.query.merge(n: {self.mapped_label_names => find_attributes})
+          .on_create_set(n: on_create_attributes)
+      end
+
       def on_create_props(find_attributes)
         find_attributes.merge(self.new(find_attributes).props_for_create)
       end
 
-      def on_match_props
-        {}.tap { |props| props[:updated_at] = DateTime.now.to_i if attributes_nil_hash.key?('updated_at'.freeze) }
+      def on_match_props(match_attributes)
+        match_attributes.tap { |props| props[:updated_at] = DateTime.now.to_i if attributes_nil_hash.key?('updated_at'.freeze) }
```

Comment:
I can drop the `freeze`, no problem.

To be clear about the "`DateTime` and this will screw them up" part; this is the same method that was being called before in the `merge`. The only change is `{}` :arrow_right: `match_attributes` instead. Functionally it's the same with regard to `updated_at`, correct?